### DISCUSSION TITLE
What frameworks/targets to include?

### DISCUSSION THREAD

**Discussion by popey** [makealinuxapp/makealinuxapp#7] [General]
My initial thoughts are that the site should focus on desktop and laptop users. I don't believe there is room here for command-line utilities, shell scripts, or server daemons. I would rather we focus on desktop and laptop applications. 

We should (IMO) focus on someone with a Linux desktop who wants to build an application or has some time and wants to do *something*, where 'building an application' could be that *thing*. 

Within that, there's the question of which languages and targets to go for. My feeling is we should signpost those places that are established such as GNOME, KDE, Flutter & Electron as communities of developers, with good documentation. We shouldn't be duplicating all of their docs, but kickstart new people and point them in the right direction.

As for languages, toolkits and frameworks, I don't have especially strong feelings here. But I do feel we should be including modern, popular and well-supported languages. So it makes sense to talk about GTK & Rust, KDE and C++, Flutter & Dart and Electron & Javascript.

But perhaps makes less sense to talk about Haskell, Perl or Fortran. (don't @ me) :D

Keen to know what others think.

**Comment by razzeee** [comment]
Hesitant to recommend flutter, but mostly due to packaging woes.

**Comment by DanielDewberry** [comment]
Hi Popey

> I don't believe there is room here for command-line utilities, shell scripts, or server daemons. I would rather we focus on desktop and laptop applications.

I think it would be premature to exclude CLI applications since they are the "bread & butter" of Linux and have been for 30 years. When written well, they form a very useful part of our non-server ecosystem. If you decide it is worth adding a CLI section then it would be useful to include a few resources such as:

- C++ [Cli11](https://github.com/CLIUtils/CLI11)
- Rust [Clap](https://docs.rs/clap/latest/clap/) 
- Python [argparse](https://docs.python.org/3/library/argparse.html)

Best regards

Daniel
